addons/delivery/views/report_shipping.xml

```diff
@@ -21,5 +21,15 @@
                 </td>
             </xpath>
         </template>
+        <template id="report_shipping3" inherit_id="stock.report_delivery_document">
+            <xpath expr="//th[@name='td_sched_date_h']" position="after">
+                <th><strong>Weight</strong></th>
+            </xpath>
+            <xpath expr="//td[@name='schedule_date']" position="after">
+                <td>
+                    <span t-field="o.weight"/>
+                </td>
+            </xpath>
+        </template>
```

Comment:
accessing the internals of the record like _columns is not a good idea... inheritance by the module that installed `weight` field was right